### PR TITLE
Add separate proposal content type for ris (opengever.ris.proposal)

### PR DESCRIPTION
**Pull request by deiferni**
With this PR we add a new separate content type for proposals (for RIS), `opengever.ris.proposal`. This will be the base for migrating proposals currently kept only in RIS back to a GEVER dossier. This approach was discussed and agreed upon in https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/1407975425/SPV+RIS+Massnahmen+zu+Kundenfeedback. Also see [HG-4265](https://4teamwork.atlassian.net/browse/HG-4265) for more details.

The corresponding RIS PR can be found in https://github.com/4teamwork/ris/pull/3334.

**Design thoughts:**
- We create a new package `opengever.ris` so that `opengever.meeting` can be deleted as a whole at some point in the future.
- The proposal is implemented as a dumb storage object without a lot of logic in GEVER
- The logic will be kept in RIS (as is the case with the current implementation)
- The workflow mirrors the relevant states/transitions from RIS
- All CRUD operations will be redirected to the RIS UI, RIS will update GEVER via REST-API and then redirect back to GEVER
- The RIS Base-URL can be configurable (And will act as a feature-flag). RIS entry-points are chosen so that they are not ambiguous and thus are hard-coded at the moment.
- This way we can prevent two-way communication GEVER-RIS and hopefully prevent to couple these two applications even more tightly. It also allows better Re-use of logic we already have in RIS. This is of course a treade-off with some disadvantages. See https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/1407975425/SPV+RIS+Massnahmen+zu+Kundenfeedback#M%C3%B6gliche-L%C3%B6sungsans%C3%A4tze-f%C3%BCr-Antrag-(mit-groben-Sch%C3%A4tzungen) for more.

**Features provided by this first step:**
- Feature flag so that this WIP state won't show up anywhere
- A proposal content type with a fieldset that mostly mirrors what we have in ris at [spv/models/proposal.py#L49](https://github.com/4teamwork/ris/blob/b2df6baec29101659826bc14a67b1b0421cbc92c/ris/spv/models/proposal.py#L49)
  - Some field names had to be changed for technical reasons (`creator`) while others are composite fields that will be used to link back to ris (e.g. `meeting_<title|url>`)
  - Those fields are filled/changed multiple times as per RIS domain logic and thus also use a short name
- A workflow that mirrors states and transitions that are relevant for GEVER 
  - In RIS we can skip the inbox and directly add/remove proposals from meetings
  - The GEVER workflow thus contains transitions to do so as well

**Things that are IMO out of scope or omitted onpurpose:**
- The API docs don't include the proposal (yet)
- Solr and new UI integration will have to be done separately

**Thoughts about maintainability:**
To ease the poor maintainer's live I have added debug actions for `Manager`. They should hopefully help to quickly resolve and/or debug out of sync state. Debug actions exists for the following things (`Manager` only):
  - the edit form can be used in the plone classic UI
  - every workflow-state can be reached with a transition performed in the classic ui

**Possible issues:**
- The user experience is not the best as we redirect between applications (This can be mitigated with further steps that are out of scope for this PR)
- The main disadvantage is that we don't have transactionality (As is also the case for almost anything that RIS already does with the GEVER). This is somewhat mitigated by providing tooling for `Manager`.

_Dear reviewer(s) please have a good and thorough look here, I still feel very rusty with this codebase and errors are very likely!_

Definition of Done: https://4teamwork.atlassian.net/wiki/spaces/CHX4TW/pages/917562/

For [HG-4265](https://4teamwork.atlassian.net/browse/HG-4265)

## Checklist

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

_Only applicable should be left and checked._

- New functionality:
  - [x] for `document` also works for `mail`
- Further improvements needed:
  - [x] Create follow-up stories and link them in the PR and Jira issue
- New translations
  - [x] All msg-strings are unicode
- Change in schema definition:
  - [x] If `missing_value` is specified, then `default` has to be set to the same value

[HG-4265]: https://4teamwork.atlassian.net/browse/HG-4265?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[HG-4265]: https://4teamwork.atlassian.net/browse/HG-4265?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: opengever/base/locales/fr/LC_MESSAGES/plone.po [hunk 1]
```diff
@@ -294,12 +294,39 @@ msgstr "Soumis"
...
+msgstr "Calendrier  (Manager: Debug)"
+
 msgid "proposal-transition-submit"
 msgstr "Soumettre"
 
+msgid "proposal-transition-submit-manage"
+msgstr "Soumettre (Manager: Debug)"
+
+msgid "proposal-transition-undecide-manage"
+msgstr "Indécider (Manager: Debug)"
```
Comment (deiferni) [inline opengever/base/locales/fr/LC_MESSAGES/plone.po:325]: Annuler décision

File: opengever/ris/locales/fr/LC_MESSAGES/opengever.ris.po [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+"Language-Name: English\n"
+"Preferred-Encodings: utf-8 latin1\n"
+"Domain: DOMAIN\n"
+"Language: fr\n"
+"X-Generator: Weblate 2.13.1\n"
+
+#. Default: "Assignee"
+#: ./opengever/ris/proposal.py
+msgid "label_assignee"
+msgstr "Personne responsable"
```
Comment (deiferni) [inline opengever/ris/locales/fr/LC_MESSAGES/opengever.ris.po:25]: Responsable

File: opengever/ris/locales/fr/LC_MESSAGES/opengever.ris.po [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+#. Default: "Business type"
+#: ./opengever/ris/proposal.py
+msgid "label_business_type"
+msgstr "Type d'affaire"
+
+#. Default: "Business URL"
+#: ./opengever/ris/proposal.py
+msgid "label_business_url"
+msgstr "URL d'affaires"
```
Comment (deiferni) [inline opengever/ris/locales/fr/LC_MESSAGES/opengever.ris.po:45]: URL de l'affaire

File: opengever/ris/locales/fr/LC_MESSAGES/opengever.ris.po [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+#. Default: "Committee title"
+#: ./opengever/ris/proposal.py
+msgid "label_committee_title"
+msgstr "titre de la commission"
```
Comment (deiferni) [inline opengever/ris/locales/fr/LC_MESSAGES/opengever.ris.po:50]: Titre

File: opengever/ris/locales/fr/LC_MESSAGES/opengever.ris.po [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+#. Default: "Description"
+#: ./opengever/ris/proposal.py
+msgid "label_description"
+msgstr "Description"
+
+#. Default: "Excerpts"
+#: ./opengever/ris/proposal.py
+msgid "label_excerpts"
+msgstr "Extraits de protocole"
```
Comment (deiferni) [inline opengever/ris/locales/fr/LC_MESSAGES/opengever.ris.po:75]: Extraits du protocole

File: opengever/dossier/menu.py [hunk 6]
```diff
@@ -14,6 +18,31 @@ class DossierPostFactoryMenu(FilteredPostFactoryMenuWithWebactions):
     def __init__(self, context, request):
         super(DossierPostFactoryMenu, self).__init__(context, request)
 
+    def __call__(self, factories):
+        factories = super(DossierPostFactoryMenu, self).__call__(factories)
+        manager_factories = []
+        for factory in factories:
```
Comment (buchi) [inline opengever/dossier/menu.py:24]: Hmm. I don't really like this approach. But anything else would require refactoring how the factory menu is computed.

File: opengever/ris/tests/test_proposal_forms.py [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+                Builder('ris_proposal')
+                .within(self.dossier)
+                .having(document=self.document)
+            )
+
+    @browsing
+    def test_add_form_is_disabled(self, browser):
+        self.login(self.manager, browser)
+        # This causes an infinite redirection loop between ++add++ and
+        # reqiure_login. By enabling exception_bubbling we can catch the
```
Comment (buchi) [inline opengever/ris/tests/test_proposal_forms.py:28]:
```suggestion
        # require_login. By enabling exception_bubbling we can catch the
```

File: opengever/ris/tests/test_proposal_forms.py [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+        with self.assertRaises(Unauthorized):
+            # with browser.expect_unauthorized():
+            browser.open(self.dossier,
+                         view='++add++opengever.ris.proposal')
+
+    @browsing
+    def test_edit_form_is_disabled(self, browser):
+        self.login(self.manager, browser)
+        # This causes an infinite redirection loop between edit and
+        # reqiure_login. By enabling exception_bubbling we can catch the
```
Comment (buchi) [inline opengever/ris/tests/test_proposal_forms.py:40]:
```suggestion
        # require_login. By enabling exception_bubbling we can catch the
```

**Review by buchi — APPROVED** [review]
👍🏻 LGTM

File: opengever/base/locales/de/LC_MESSAGES/plone.po [hunk 9]
```diff
@@ -294,12 +294,39 @@ msgstr "Eingereicht"
...
 msgstr "Einreichen"
 
+msgid "proposal-transition-submit-manage"
+msgstr "Einreichen (Manager: Debug)"
+
+msgid "proposal-transition-undecide-manage"
+msgstr "Beschluss entfernen (Manager: Debug)"
+
+msgid "proposal-transition-unschedule-manage"
+msgstr "Traktandum streichen  (Manager: Debug)"
```
[the remark this answers is not in the document]
Comment (deiferni) [reply opengever/base/locales/de/LC_MESSAGES/plone.po]: solved in the meantime 😉

**Pull request merged**